MekHQ/src/mekhq/campaign/parts/Armor.java

```diff
@@ -572,31 +572,44 @@ public boolean isEnoughSpareArmorAvailable() {
         return getAmountAvailable() >= amountNeeded;
     }
 
+    /**
+     * Searches the warehouse for a compatible parts and returns how many points of armor are found.
+     * @return returns points of armor are found
+     */
     public int getAmountAvailable() {
-        Armor a = (Armor) campaign.getWarehouse()
-                                .findSparePart(part -> (part instanceof Armor) &&
-                                                             part.isPresent() &&
-                                                             !part.isReservedForRefit() &&
-                                                             isSameType((Armor) part));
+        return campaign.getWarehouse()
+                     .streamSpareParts()
+                     .mapToInt(part -> ((part instanceof Armor armor) &&
+                                              armor.isPresent() &&
+                                              !armor.isReservedForRefit() &&
+                                              isSameType(armor)) ? armor.getAmount() : 0)
+                     .sum();
+    }
+
+    /**
+     * Searches the warehouse for a compatible parts changes the amount available. This will continue to remove parts
+     * if more than one part needs to be removed to cover the amount.
+     * @param amount points of armor to add or remove
+     */
+    public void changeAmountAvailable(int amount) {
+        if (amount == 0) {
+            return;
+        }
 
-        return (a == null) ? 0 : a.getAmount();
-    }
+        int amountRemaining = amount;
+        int priorAmount = 0;
 
-    public void changeAmountAvailable(int amount) {
-        Armor a = (Armor) campaign.getWarehouse()
-                                .findSparePart(part -> (part instanceof Armor) &&
-                                                             part.isPresent() &&
-                                                             Objects.equals(getRefitUnit(), part.getRefitUnit()) &&
-                                                             isSameType((Armor) part));
-
-        if (null != a) {
-            a.setAmount(a.getAmount() + amount);
-            if (a.getAmount() <= 0) {
-                campaign.getWarehouse().removePart(a);
-            }
-        } else if (amount > 0) {
-            campaign.getQuartermaster()
-                  .addPart(new Armor(getUnitTonnage(), type, amount, -1, false, isClanTechBase(), campaign), 0);
+
+        while ((amountRemaining != 0) && priorAmount != amountRemaining) {
+            priorAmount = amountRemaining;
+
+            amountRemaining = changeAmountAvailableSingle(amountRemaining);
+        }
+
+        if (amountRemaining > 0) {
+            logger.error("Still trying to add armor but that shouldn't have been a problem!");
+        } else if (amount < 0) {
+            logger.error("Still trying to remove armor but no more armor is in the warehouse!");
```

Comment:
logger.error should be restricted to only when you have an actual error that you can do something to fix, either because something really wrong happened, or because some data is corrupted, either is not the case, if this is a log that just works as a kind of `printf("<<<<<< HERE!!!")` then you should change the log level to `debug` or `trace`.